packages/gestalt/src/TableRowDrawer.js

```diff
@@ -54,7 +54,7 @@ export default function TableRowDrawer({ children, drawerContents, id }: Props):
       </tr>
       <tr id={id}>
         <td className={styles.drawer} colSpan={Children.count(children) + 1}>
-          <Box padding={2}>{drawerContents}</Box>
+          <Box padding={drawerContents ? 2 : 0}>{drawerContents}</Box>
```

Comment:
If there is no drawer content, should this even be rendered?

Reply:
@ayeshakmaz Indeed. Good observation.

DONE